fluent_pose_synthesis/data/map_gloss_to_pose.py

```diff
@@ -0,0 +1,84 @@
+from typing import Dict, List, Tuple, Any
+import tensorflow as tf
+
+
+def create_gloss_to_pose_dict(dgs_types_dataset: tf.data.Dataset) -> Tuple[Dict[str, Dict[str, Any]], Dict[str, List[Dict[str, Any]]]]:
+    """
+    Generate a mapping of glosses to their corresponding pose data from DGS Types dataset.
+    """
+    gloss_to_pose_dict = {}
+    gloss_pose_conflicts = {}
+    
+    for datum in dgs_types_dataset["train"]:
+        # Get glosses and id
+        glosses = datum['glosses'].numpy().tolist()
+        datum_id = datum['id'].numpy().decode('utf-8')
+        is_galex = datum_id.startswith('galex_')
+        
+        # Get pose (if exists)
+        pose = datum['views']['pose'].numpy().tolist()
+        pose = pose[0] if pose else None
+                
+        # Keep complete original data
+        original_data = {k: v for k, v in datum.items()}
+        
+        # Add pose for each gloss
+        for gloss in glosses:
+            # Convert gloss to string if it's bytes
+            gloss = gloss.decode('utf-8') if isinstance(gloss, bytes) else gloss
+            
+            # Check if a pose already exists for this gloss
+            if gloss in gloss_to_pose_dict:
+                existing_pose = gloss_to_pose_dict[gloss]['views']['pose']
+                existing_id = gloss_to_pose_dict[gloss]['id'].numpy().decode('utf-8')
+                existing_is_galex = existing_id.startswith('galex_')
+                
+                # If existing pose is None, update with new pose
+                if existing_pose is None and pose is not None:
+                    gloss_to_pose_dict[gloss]['views']['pose'] = pose
+                    continue
```

Comment:
instead of `continue`, make use of the `elif` here. and then before `# Create new entry` you can `else`
that would remove all need for `continue`